gadget/install/partition_test.go

```diff
@@ -216,6 +216,32 @@ var mockOnDiskStructureWritableAfterSave = gadget.OnDiskStructure{
 	Size: 2*quantity.SizeGiB + 717*quantity.SizeMiB + 1031680,
 }
 
+// mustLayOutVolumeFromGadget takes a gadget rootdir and lays out the
+// partitions as specified. This function does not handle multiple volumes and
+// is meant for test helpers only. For runtime users, with multiple volumes
+// handled by choosing the ubuntu-* role volume, see LaidOutSystemVolumeFromGadget
+func mustLayOutVolumeFromGadget(c *C, gadgetRoot string, model gadget.Model) (*gadget.LaidOutVolume, error) {
```

Comment:
why do we need this function? it's because the test gadget don't have the right roles?

Reply:
yes lots of the test gadgets have no system-boot roles, etc. because they are just testing the math behind the functions for gadget layouts and such and I didn't want to refactor all of those tests